keystore/src/transaction.rs

```diff
@@ -0,0 +1,340 @@
+            Entity::EpochEncryptionKeyPair(mls_epoch_encryption_key_pair) => {
+                mls_epoch_encryption_key_pair.mls_save(tx).await
+            }
+            Entity::MlsCredential(mls_credential) => mls_credential.mls_save(tx).await,
+            Entity::PersistedMlsGroup(persisted_mls_group) => persisted_mls_group.mls_save(tx).await,
+            Entity::PersistedMlsPendingGroup(persisted_mls_pending_group) => {
+                persisted_mls_pending_group.mls_save(tx).await
+            }
+            Entity::MlsPendingMessage(mls_pending_message) => mls_pending_message.mls_save(tx).await,
+            Entity::E2eiEnrollment(e2ei_enrollment) => e2ei_enrollment.mls_save(tx).await,
+            Entity::E2eiRefreshToken(e2ei_refresh_token) => e2ei_refresh_token.mls_save(tx).await,
+            Entity::E2eiAcmeCA(e2ei_acme_ca) => e2ei_acme_ca.mls_save(tx).await,
+            Entity::E2eiIntermediateCert(e2ei_intermediate_cert) => e2ei_intermediate_cert.mls_save(tx).await,
+            Entity::E2eiCrl(e2ei_crl) => e2ei_crl.mls_save(tx).await,
+        }
+    }
+
+    async fn handle_entity_id(tx: &TransactionWrapper<'_>, entity_id: &EntityId) -> CryptoKeystoreResult<()> {
+        use crate::entities::EntityMlsExt;
+        match entity_id {
+            id @ EntityId::SignatureKeyPair(_) => MlsSignatureKeyPair::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::HpkePrivateKey(_) => MlsHpkePrivateKey::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::KeyPackage(_) => MlsKeyPackage::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::PskBundle(_) => MlsPskBundle::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::EncryptionKeyPair(_) => MlsEncryptionKeyPair::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::EpochEncryptionKeyPair(_) => MlsEpochEncryptionKeyPair::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::MlsCredential(_) => MlsCredential::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::PersistedMlsGroup(_) => PersistedMlsGroup::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::PersistedMlsPendingGroup(_) => PersistedMlsPendingGroup::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::MlsPendingMessage(_) => MlsPendingMessage::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::E2eiEnrollment(_) => E2eiEnrollment::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::E2eiRefreshToken(_) => E2eiRefreshToken::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::E2eiAcmeCA(_) => E2eiAcmeCA::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::E2eiIntermediateCert(_) => E2eiIntermediateCert::mls_delete(tx, id.as_id()).await,
+            id @ EntityId::E2eiCrl(_) => E2eiCrl::mls_delete(tx, id.as_id()).await,
+        }
+    }
+
+    #[cfg(target_family = "wasm")]
+    fn scan_collection_name(&self) -> &'static str {
```

Comment:
Indexed DB needs the name of the collections that will be affected in the transaction before hand (🤯), so had to implement this exclusively for WASM.